frontend/src/components/visual-editor/v3/components/ReactFlowWrapper.tsx

```diff
@@ -0,0 +1,268 @@
+    const sourceNode = getBlockFromCache(sourceNodeId);
+    const payload: Partial<IBlockAttributes> =
+      sourceHandle === "nextBlocks"
+        ? {
+            nextBlocks: [
+              ...(sourceNode?.nextBlocks || []).filter(
+                (id) => id !== targetNodeId,
+              ),
+              targetNodeId,
+            ],
+          }
+        : { attachedBlock: targetNodeId };
+
+    setEdges((eds) => addEdge({ ...params, type: "buttonedge" }, eds));
+
+    onMoveNode({
+      id: sourceNodeId,
+      ...payload,
+    });
+  };
+  const handleNodeClick: NodeMouseHandler<Node> = useCallback(
+    (e, { id }) => {
+      if (!selectedNodeIds.includes(id)) {
+        if (selectedCategoryId && !e.ctrlKey) {
+          updateVisualEditorURL(selectedCategoryId, id);
+          setSelectedNodeIds([id]);
+        } else {
+          setSelectedNodeIds((nodes) => [...nodes, id]);
+        }
+      }
+    },
+    [
+      selectedCategoryId,
+      selectedNodeIds,
+      setSelectedNodeIds,
+      updateVisualEditorURL,
+    ],
+  );
+  const handleNodeDoubleClick: NodeMouseHandler<Node> = useCallback(
+    (_, { id }) => {
+      if (selectedNodeIds.length === 1) {
+        onNodeDoubleClick?.(id) || openEditDialog(id);
+      }
+    },
+    [selectedNodeIds.length, onNodeDoubleClick, openEditDialog],
+  );
+  const handleNodeDragStart: OnNodeDrag<Node> = useCallback(
+    (_, { id }) => {
+      if (selectedCategoryId && selectedNodeIds.length === 1) {
+        setSelectedNodeIds([id]);
+        updateVisualEditorURL(selectedCategoryId, id);
+      }
+    },
+    [
+      selectedNodeIds,
+      selectedCategoryId,
+      setSelectedNodeIds,
+      updateVisualEditorURL,
+    ],
+  );
+  const handleNodeDragStop: OnNodeDrag<Node> = useCallback(
+    async (_e, _node, nodes) => {
+      try {
+        for (const { id, position } of nodes) {
+          onMoveNode({
+            id,
+            position,
+          });
+        }
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.error("Unable to update node", e);
+      }
+    },
+    [onMoveNode],
+  );
+
+  useOnViewportChange({
+    onEnd: onViewport,
+  });
+
+  useEffect(() => {
+    if (deleteKeyPressed && selectedNodeIds.length) {
+      openDeleteManyDialog(selectedNodeIds).then((confirm) => {
+        if (confirm) {
+          removeBlockIdParam();
+        }
+      });
+    }
+  }, [
+    onDeleteNodes,
+    selectedNodeIds,
+    deleteKeyPressed,
+    openDeleteManyDialog,
+    removeBlockIdParam,
+  ]);
+
+  useEffect(() => {
+    setViewport(defaultViewport);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [nodesInitialized]);
+
+  return (
+    <ReactFlow
+      defaultEdges={defaultEdges}
+      edges={defaultEdges}
+      defaultNodes={defaultNodes}
+      nodes={defaultNodes}
+      defaultViewport={defaultViewport}
+      maxZoom={4}
+      minZoom={-2}
+      nodeTypes={NODE_TYPES as any}
```

Comment:
```suggestion
      nodeTypes={NODE_TYPES}
```

Reply:
By removing the any cast we face this TS error 
<img width="207" height="69" alt="image" src="https://github.com/user-attachments/assets/15d3d225-84ee-420b-9ff6-93c52449be61" />